### ISSUE TITLE
Error when using `nvim_buf_call` with `edit` on hidden buffer with leading parentheses

### ISSUE THREAD

**Issue by niqodea** [neovim/neovim#25877]
### Problem

When executing :lua vim.api.nvim_buf_call(n, function() vim.cmd("edit") end) where buffer n is hidden and contains a matched parenthesis as the first non-space character, Neovim throws the following error:

```
E5108: Error executing lua [string ":lua"]:1: Error executing lua: vim/_editor.lua:0: nvim_exec2()..BufWinEnter Autocommands for "*"..function <SNR>12_Highlight_Matching_Pair, line 149: Vim(call):E801: ID already taken: 3
```

This issue seems to occur during the `BufWinEnter` autocommands, specifically within a function responsible for highlighting matching pairs.

### Steps to reproduce

- Run `nvim -u NORC myfile`
- Insert `[]` in the first line and save
- Switch to a new buffer: `enew`
- Run the following Ex command: `lua vim.api.nvim_buf_call(1, function() vim.cmd("edit") end)`
- Run again the previous Ex command

### Expected behavior

The expected behavior is that the hidden buffer is edited without any errors, even if it starts with a parenthesis.

### Neovim version (nvim -v)

NVIM v0.9.0

### Vim (not Nvim) behaves the same?

NA

### Operating system/version

Ubuntu 22.04.3 LTS

### Terminal name/version

Windows Terminal

### $TERM environment variable

tmux-256color

### Installation

Build from repo

**Comment by zeertzjq** [comment]
The steps to reproduce are a bit vague:
- What is "a minimal configuration"? Please provide a configuration that can be sourced with `nvim --clean -u`.
- What is "a matched parenthesis"? An opening parenthesis or a closing one?

**Comment by niqodea** [comment]
Sorry @zeertzjq, I updated the steps, should be much clearer now

**Comment by zeertzjq** [comment]
I think this has already been fixed by #25734.
